### ISSUE TITLE
Usermin / OS X / Safari - problem sending mail

### ISSUE THREAD

**Issue by konamac** [qooob/authentic-theme#1102]
Authentic Theme 19.12 / Webmin 1.881 / Virtualmin 6.02 / Usermin 1.741 / Centos 7.4.1708

When sending mail using the "Send Mail" button on the Usermin webmail  'Compose Email' screen, the interface freezes at the mail sending page (/mailbox/send_mail.cgi); the red bar animation at the top of the page keeps bouncing back and forth, suggesting activity, but the operation eventually times out and the mail is not sent.

A customer reported this on "latest Mac, latest Safari" and I can duplicate the issue on OS X 10.12.6, Safari 11.1.

Cheers!

**Comment by qooob** [comment]
Thanks for reporting, I'll try to reproduce it.

**Comment by konamac** [comment]
PS On the same OS X machine, I can successfully send email using both Chrome and Firefox; it's a Safari-specific issue.

**Comment by konamac** [comment]
Thanks!

**Comment by qooob** [comment]
Unfortunately, it works just fine on my _MacBook Air_ with Safari 11.0.3.

The mail is being reworked now. Did this issue start happening recently, or it's been like this forever for your customer?

**Comment by konamac** [comment]
It was just reported yesterday, shortly after I updated to the latest Usermin and then to the latest Authentic (before that, IIRC I was still on Authentic 19.09.2 as I was holding off the theme update until the latest Usermin update came through on the Centos repos.)

**Comment by konamac** [comment]
May be a Centos-specific issue?

**Comment by qooob** [comment]
Could you open console in Safari and see what happens in Network tab, after you submit the mail?

**Comment by konamac** [comment]
It attempts to load sendmail.cgi;  this times out after 60 seconds with the error "Failed to load resource: the server responded with a status of 400 (Timeout)".  After this, it successfully loads index.cgi.

**Comment by konamac** [comment]
(Are there any particular server-side logs you'd like me to check?)

**Comment by qooob** [comment]
What happens if you switch back to Gray Theme?

I don't use sendmail, this is the reason why it works for me. I'll need to do more tests.

Switch themes and try again please.

**Comment by konamac** [comment]
Also FYI, just checked and the last time the user successfully sent mail was "5 or 6 days ago" - so before the recent update.
Will try with Gray theme.

**Comment by konamac** [comment]
The problem disappears if I use "Gray Framed Theme" - send_mail.cgi loads successfully, and the mail is sent.

**Comment by konamac** [comment]
However loading Gray Framed Theme makes me appreciate your beautiful theme all the more ;)

**Comment by qooob** [comment]
Wait, no `send_mail.cgi` is also used with _Postfix_. Okay. Then I have no idea what's happening.

While it's _Usermin_ and it has limited access, could you create for me a temporary account, so I could login an see what's going on?

Do you have XMPP, Slack or Wire where we could chat and share data securely?

**Comment by qooob** [comment]
Can you do the following:

```
yum downgrade usermin
```

Clear cache and try again.

Then go to Webmin and upgrade the theme using Theme Configuration, to latest development version.

Then try again. Do you still have an issue?

**Comment by qooob** [comment]
For your information, I'm having issue on Debian but not on Fedora.

**Comment by konamac** [comment]
Just back from the gym, sounds like you made more progress than I did ;)
Glad you can duplicate it, irritating that it's varying by distro!

**Comment by qooob** [comment]
In that case, I don't see how it's the theme issue.

**Comment by qooob** [comment]
But I'll investigate more and in case it's the distro I will let guys know.

**Comment by konamac** [comment]
I agree it is really weird that it's only happening in Safari if it's a server-side issue!

**Comment by qooob** [comment]
😳 After I installed it on Debian (only Usermin), it didn't load even on Chrome.

It needs more testing. I'll do it a bit later.

**Comment by konamac** [comment]
OK, thanks again for getting on this so promptly!

**Comment by qooob** [comment]
I have logged in using _Safari_ to production server and _Usermin 1.841_ installed from _Webmin_ repo - no issue when sending mail. I'm going to diff two version from different repos to figure things out.

Meanwhile, can you just run 

```
yum reinstall http://download.webmin.com/download/yum/usermin-1.741-1.noarch.rpm
```

It will install the same version of _Usermin_ but from _Webmin_ repo.

Give it a try please. It's safe to do.

**Comment by konamac** [comment]
Hiya, I've re-installed usermin as suggested, and I've emptied the caches in Safari, but I'm still seeing the same problem.  (The same failure occurs when using the "Save and Edit" button on the Compose Mail screen - which is also invoking send_mail.cgi.)

Are you able to test with a later version of Safari ?

**Comment by qooob** [comment]
Yes, it was the latest version of Safari. At first part of the day, I had previous version (not sure which one was that) and later, I received big update and tried it again at night, after installing it - all was fine.

Okay, I am afraid I'm gonna need a login. Is that possible? It's not clear what is broken. Guessing could work but it's time consuming.

In the morning I'm going to be busy, with ballet concert for my daughter but later in a day, I could login and see what's going on.

So if you downgrade, like suggested before, what happens? If you downgrade and it works, can you force update the theme (ignoring warning) to the latest development version and see if it would work with previous version of Usermin and latest theme?

**Comment by konamac** [comment]
With typically inconvenient timing, I'm actually busy with school holidays for much of the next week myself!  I'll set up an email account with usermin login for you and send the details via your website this morning;  more detailed changes to the server (which is a production server, so I have to be cautious) I can look at after next week.  Thanks and enjoy the ballet!

**Comment by qooob** [comment]
I works in Chrome. I'm very surprised. Did you try to force update the theme to the latest development version? Please do it. I'll be able to run tests for a little bit more today still.

**Comment by qooob** [comment]
Okay, I have done some more tests.

The only difference that I see in headers between Gray Theme and Authentic Theme, the accept header that is set in Gray Theme to:

```
Accept: text/html,application/xhtml+xml,application/xml
```

and in Authentic Theme it's set to:

```
Accept: text/html
```
-----

I'm reworking mail anyway right now and this issue might be self-resolved in the process of rework. I have also reproduced it in my Safari with my server, so you can delete my test account.

Thank you for reporting.

**Comment by konamac** [comment]
Thanks again!

**Comment by qooob** [comment]
[Additional ref.:](https://www.virtualmin.com/node/55547) Mailbox not refreshing using Authentic Theme.
